### ISSUE TITLE
Calendso doesn't check multiple Google calendars for conflicts...

### ISSUE THREAD

**Issue by crawfordifland** [calendso/calendso#97]
I use my personal Gmail account to see and manage my Google calendar. I've shared my G Suite-based work calendars (with edit permissions) with my Gmail account so that I can see and manage everything in one place. 

I've noticed that if I give Calendso permission to access my Gmail account and create calendar events, it only checks that account for conflicts when someone goes to book - it doesn't check _all calendars_ that my Gmail account has access to. This means I could have a conflict on another calendar but Calendso will still let people book me during that time, because my primary Gmail calendar doesn't have a conflict. 

Is this something that can be remedied by adding a different scope when setting up the Google Calendar API OAuth consent screen?

**Comment by baileypumfleet** [comment]
This is caused because the API call is set to check the calendar marked as primary. As far as I know, you can’t tell the API to check all calendars.

**Comment by emrysal** [comment]
The question really is how to optimise this process; but it should be possibly by first querying all calendars under a user account, and then running a bulk query on freebusy.

Alternatively, and prefered but less automatic, use the UI for the Google Calendar integration to statically define the calendars the user wants to add to his integration, store this as part of the configuration and query those calendars on every freebusy call.

**Comment by zacharykimball** [comment]
+1 for this as a valuable (in fact essential) feature.

Somehow Calendly manages to make this work, so I know it's technically possible -- on that service, you can select multiple calendars to "check for conflicts".  I've tested this with multiple personal calendars (i.e., multiple of my own calendars created on my Gmail account) and also with shared calendars (i.e., calendars that have been shared with me but are *not* owned by my Gmail account).  Both configurations work.

Not sure *how* exactly it the actual implementation works (OAuth etc) but it seemed to be a pretty seamless process there. 
 Sharing screenshot below (the options are at https://calendly.com/app/calendar_connections).
![image](https://user-images.githubusercontent.com/4469173/116145073-0c82b880-a69a-11eb-9a6d-594bc667dff7.png)

**Comment by JohnKimDev** [comment]
+1 for this MUST-HAVE feature

I believe this is technically a possible solution. 

If you look at this code 
```
calendar.freebusy.query({
    requestBody: {
        timeMin: dateFrom,
        timeMax: dateTo,
        items: [ {
            "id": "primary"
        } ]
    }
}, (err, apires) => {
    if (err) {
        reject(err);
    }
    resolve(apires.data.calendars.primary.busy)
});
```
https://github.com/calendso/calendso/blob/6091f7ba86afa796595071dd4eaf3e80e71add8f/lib/calendarClient.ts#L137

Currently, the code is only looking for the (hardcoded) "primary" ID of Google calendar only. According to the Google Calendar API spec, https://developers.google.com/calendar/v3/reference/freebusy/query
`items` takes the list of IDs (calendar IDs). 
 
**The solution is to retrieve the list of calendars BEFORE passing the IDs into this `freebusy` query .**
https://developers.google.com/calendar/v3/reference/calendarList/list

FYI, this requires the calendar list view permission, I don't think the currently calendso doc suggests enabling this permission during the installation.

**Comment by jtylerhartley** [comment]
This should be fixed with PR #147. Major props to @jfernandogt for implementing this!
